schema/generator.go

```diff
@@ -174,7 +174,14 @@ func (g *Generator) generateDDLs(desiredDDLs []DDL) ([]string, error) {
 			}
 
 			// Column is obsoleted. Drop column.
-			ddl := fmt.Sprintf("ALTER TABLE %s DROP COLUMN %s", g.escapeTableName(desiredTable.name), g.escapeSQLName(column.name))
+			var ddl string
+			switch g.mode {
+			case GeneratorModeMssql:
+				ddls = g.removeColumnConstraints(currentTable, column.name, ddls)
+				ddl = fmt.Sprintf("ALTER TABLE %s DROP COLUMN %s", g.escapeTableName(desiredTable.name), g.escapeSQLName(column.name))
+			default:
+				ddl = fmt.Sprintf("ALTER TABLE %s DROP COLUMN %s", g.escapeTableName(desiredTable.name), g.escapeSQLName(column.name))
```

Comment:
It seems like L181 and L183 are the same thing and you don't need to branch them.